ibm/data_source_ibm_is_floating_ip.go

```diff
@@ -0,0 +1,168 @@
+package ibm
+
+import (
+	"fmt"
+
+	"github.com/IBM/vpc-go-sdk/vpcclassicv1"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	floatingIPName    = "name"
+	floatingIPAddress = "address"
+	floatingIPStatus  = "status"
+	floatingIPZone    = "zone"
+	floatingIPTarget  = "target"
+	floatingIPTags    = "tags"
+)
+
+func dataSourceIBMISFloatingIP() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMISFloatingIPRead,
+
+		Schema: map[string]*schema.Schema{
+
+			floatingIPName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the floating IP",
+			},
+
+			floatingIPAddress: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Floating IP address",
+			},
+
+			floatingIPStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Floating IP status",
+			},
+
+			floatingIPZone: {
+				Type:          schema.TypeString,
+				Computed:      true,
+				ConflictsWith: []string{isFloatingIPTarget},
+				Description:   "Zone name",
+			},
+
+			floatingIPTarget: {
+				Type:          schema.TypeString,
+				Computed:      true,
+				ConflictsWith: []string{isFloatingIPZone},
+				Description:   "Target info",
+			},
+
+			floatingIPTags: {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Set:         resourceIBMVPCHash,
+				Description: "Floating IP tags",
+			},
+		},
+	}
+}
+
+func dataSourceIBMISFloatingIPRead(d *schema.ResourceData, meta interface{}) error {
+	userDetails, err := meta.(ClientSession).BluemixUserDetails()
+	if err != nil {
+		return err
+	}
+	floatingIPName := d.Get(isFloatingIPName).(string)
+	if userDetails.generation == 1 {
+		err := classicFloatingIPGet(d, meta, floatingIPName)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := floatingIPGet(d, meta, floatingIPName)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func classicFloatingIPGet(d *schema.ResourceData, meta interface{}, name string) error {
+	sess, err := classicVpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	floatingIPOptions := &vpcclassicv1.ListFloatingIpsOptions{}
```

Comment:
Use the option start parameter....which we added for other datasources for List
```
start
string

A server-supplied token determining what resource to start the page on
```
check other datasources